.pre-commit-config.yaml

```diff
@@ -0,0 +1,9 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
```

Comment:
Would it be possible to keep this whole file in the `python` folder?

Reply:
pre-commit installs hooks in a completely isolated environment that does not affect global packages. If other contributors don't touch the `python` folder, the `black` hook won't even run at all.

Reply:
I tried putting `.pre-commit-config.yaml` under the `python` directory, but it won't allow any other `.pre-commit-config.yaml` to be used if anyone decides to have one under `java/` or any other folder.
Seems like a limitation of pre-commit.
So the only way is to keep it on the root dir, and explicitly whitelist the dirs that each hook should handle.